routes/gardenRoutes.js

```diff
@@ -0,0 +1,47 @@
+var express = require('express');
+var garden = require('../garden');
+var bodyparser = require('body-parser');
+
+var router = express.Router();
+
+// for body-parser, request.body's field names are the same as input form names
+router.route('/')
+  .get(function(request, response) {
+    response.status(200).json(garden.getPlants());
+  })
+  .post(function(request, response) {
+    var newPlant = request.body;
+
+    if (!newPlant) {
+      response.status(400).json('problem with adding plant');
+    } else {
+      newBook = garden.addPlant(newPlant);
+      response.status(201).send(newPlant);
+    }
+  });
+
+router.route('/#planting')
```

Comment:
The server isn't going to route on # routes like this. This is why # was used for client-side routing until pushState was available (because the server would ignore it). If you wanted to use pushState on the client, then you have to set your server up in such a way that it will still work if loading a page where only client-side routes are supported (such as /planting). That's what the spa-server-route in the food-truck-app example did. It has to be used after any other routing is specified in your server, however, or it will handle requests you don't want it to.